include/cutlass/gemm/collective/xe_mma_w8a8.hpp

```diff
@@ -235,11 +229,11 @@ struct CollectiveMma<MainloopIntelW8A8<Stages, Schedule>, TileShape_, ElementA_,
     Tensor tCgA = thr_mma.partition_A(gA);
     Tensor tCgB = thr_mma.partition_B(gB);
 
-    Tensor tCrA = make_tensor<uint8_t>(make_fragment_layout(mainloop.tiled_copy_a, tCgA(_,_,_,0).shape()));
-    Tensor tCrB = make_tensor<uint8_t>(make_fragment_layout(mainloop.tiled_copy_b, tCgB(_,_,_,0).shape()));
+    Tensor tCrA_fp16 = make_tensor<half_t>(make_fragment_layout(mainloop.tiled_copy_a, tCgA(_,_,_,0).shape()));
+    Tensor tCrB_fp16 = make_tensor<half_t>(make_fragment_layout(mainloop.tiled_copy_b, tCgB(_,_,_,0).shape()));
 
-    Tensor tCrA_fp16 = make_fragment_like<half_t>(tCrA);
-    Tensor tCrB_fp16 = make_fragment_like<half_t>(tCrB);
+    Tensor tCrA = make_tensor(reinterpret_cast<uint8_t*>(tCrA_fp16.data()), tCrA_fp16.layout());
+    Tensor tCrB = make_tensor(reinterpret_cast<uint8_t*>(tCrB_fp16.data()), tCrB_fp16.layout());

```

Comment:
This change is aimed at using fewer registers for FP8 -> FP16 conversion.
It uses the same tensor for FP16 & FP8 elements, so elements should be converted starting from the last one.

![image](https://github.com/user-attachments/assets/25ad9818-adb3-457e-891f-9b48baed5639)

Reply:
Rolling back this change fixed the second source of regression. I didn't look into the root-cause as to why this change caused a regression but my guess is that the compiler couldn't optimize well with this change due to potential aliasing. However, I could be wrong because before #374 had landed, I remember this change had been resulting in a ~2% perf boost on BMG at @jiyang1011's end for the FP8 GEMM's example default set of input shapes.

Used DPCPP nightly dated March 24, 2025.
IGC:
<img width="882" alt="image" src="https://github.com/user-attachments/assets/aa4473bc-25f4-47fd-a0c7-299325508d82" />